### PR TITLE
introducing latest blog post links to articles

### PR DESCRIPTION
**Pull request by JiggyPete**
This functionality is behind the feature flag: 'latest_blog_post_links'

### CODE REVIEW

File: spec/lib/core/entity/article_spec.rb [hunk 1]
```diff
@@ -124,6 +124,64 @@ module Core
...
+                'path' => '/oldest-post',
+                'date' => '2015-01-01'
+              }
+            ]
+          }
+        end
+
+        it 'has 2 article links' do
+          expect(subject.latest_blog_post_links.length).to eq(2)
+          expect(subject.latest_blog_post_links.first).to be_an_instance_of(ArticleLink)
```
Comment (baphled) [inline spec/lib/core/entity/article_spec.rb]: I'd probably split these into two different specs .... Somthing like 'has 2 articles' and 'all links are ArticleLink' :)

File: app/views/articles/_article_links.html.erb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<h2 class="related-links__heading"><%= title %></h2>
+
+<div class="related-links">
+  <ul class="unstyled-list">
+    <% article_links.each do |article_link| %>
+      <li class="related-links__link">
+        <%= render article_link %>
+        <% if article_link.date.present? %>
+          <div>
+            <%= article_link.date.strftime("#{article_link.date.day.ordinalize} %B %Y") %>
```
Comment (tomas-stefano) [inline app/views/articles/_article_links.html.erb]: I don't know the code standards of the frontend project, but I think here you can use the I18n.l and move the format to a locale.

Comment (JiggyPete) [inline app/views/articles/_article_links.html.erb]: We considered that, and left it be as it was only used in the one place

File: spec/helpers/articles_helper_spec.rb [hunk 3]
```diff
@@ -1,25 +1,29 @@
-RSpec.describe ArticlesHelper, '#category_has_more_content?', type: :helper do
-  let(:article)                  { double(categories: [category]) }
-  let(:category_id)              { 'category-id' }
-  let(:category)                 { Core::Category.new(category_id, contents: contents) }
-  let(:contents)                 { Array.new(4) { double } }
-  let(:related_content_category) { Core::Category.new(category_id, contents: contents) }
+RSpec.describe ArticlesHelper, type: :helper do
```
Comment (neoeno) [inline spec/helpers/articles_helper_spec.rb]: I know I made this change — but is it needed in this PR since we didn't end up modifying the helper?

Comment (JiggyPete) [inline spec/helpers/articles_helper_spec.rb]: you're the best person to answer that :-)

Comment (neoeno) [inline spec/helpers/articles_helper_spec.rb]: I'm gonna say it's not needed :+1:

Comment (JiggyPete) [inline spec/helpers/articles_helper_spec.rb]: having looked into it, I'd say you're tidy up is neater.

Comment (neoeno) [inline spec/helpers/articles_helper_spec.rb]: I think so too ;) — but not really related to this PR.

File: spec/lib/core/entity/article_spec.rb (hunk 1 above)
Comment (neoeno) [inline spec/lib/core/entity/article_spec.rb]: [agreed](https://github.com/moneyadviceservice/frontend/pull/867/files#diff-1382fe65be6771aad9f32c61229413d6R146) :)

File: lib/core/entity/article_link.rb [hunk 4]
```diff
@@ -3,10 +3,12 @@ class ArticleLink
     include ActiveModel::Conversion
     attr_reader :title
     attr_reader :path
+    attr_reader :date
 
-    def initialize(title, path)
+    def initialize(title, path, date = nil)
       @title = title
       @path = path
+      @date = Date.parse date if date.present?
```
Comment (benlovell) [inline lib/core/entity/article_link.rb]: `if date` would suffice here. Can you also put parens on the `parse` call.

File: spec/lib/core/entity/article_spec.rb [hunk 5]
```diff
@@ -124,6 +124,67 @@ module Core
...
+      context 'no latest_blog_post_links' do
+        let(:related_content) { {} }
+
+        it 'results in empty list' do
+          expect(subject.latest_blog_post_links).to be_empty
+        end
+      end
+
+      context 'empty latest_blog_post_links' do
+
```
Comment (benlovell) [inline spec/lib/core/entity/article_spec.rb]: Remove this line.

File: lib/core/entity/article_link.rb (hunk 4 above)
Comment (neoeno) [inline lib/core/entity/article_link.rb]: Both done!

**Comment by benlovell** [comment]
:+1:

**Pull request merged**